lib/elixir/lib/inspect.ex

```diff
@@ -372,6 +390,27 @@ defimpl Inspect, for: Reference do
 end
 
 defimpl Inspect, for: Any do
+  defmacro __deriving__(module, struct, options) do
+    only = Keyword.get(options, :only, [])
+    except = Keyword.get(options, :except, []) ++ [:__exception__, :__struct__]
```

Comment:
What happens if both options are given? Should we raise? Should they double filter? On `import`, giving both raises unless a meta key is used (which we don't have here).

Reply:
I had the same doubts. I followed the behaviour of `doctest`, which double filters and gives more priority to`:except`: 

https://github.com/elixir-lang/elixir/blob/c744d0be1c8dcf7b439644e9f8140b30ccb6ec4f/lib/ex_unit/lib/ex_unit/doc_test.ex#L234-L241